Interactive Console Updates

skuntz opened the pull request:
This PR provides a set of updates to the interactive debug console, object mapping support, and bug fixes.

- Ensure signal command line flags are processed before the checkpoint directory is created and have it fail immediately if the --checkpoint-prefix PREFIX is not valid. 
- Add support for user defined commands in the interactive console, consisting of a named sequence of commands and supporting nested definitions 
- Single-rank, multithreaded interactive debug console support. In multithreaded simulations, interactive console is invoked from the syncManager, starting in thread 0. The user can navigate the object map, set watchpoints, etc. within a thread. The “info” command provides the rank, thread ID, and process ID for threads and the “thread” command switches control between threads. 



----

sst-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

skuntz committed:
Interactive Console Updates
* Support for single-rank, multithreaded interactive debug console.
  Breaks into interactive console is at a sync point and is managed
  by the syncManager. It breaks into the interactive console when
  any thread has triggered the interactive console and starts in
  thread 0 initially. The user can navigate the object map,
  set watchpoints, etc. for components within the thread and switch
  between threads to access other components.
* Support for user-defined commands in the interactive debug console
  which are a named sequence of commands and can be nested.
* Updates to the parsing of --checkpoint-prefix and creation of the
  checkpoint directory. Ensure that the signal flags are handled
  before the checkpoint directory is created and fails immediately
  if the PREFIX is not valid.

sst-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

leekillough commented:
I expect a lot of merge conflicts with https://github.com/sstsimulator/sst-core/pull/1517. If it is merged soon enough, I'll help resolve the conflicts.

kpgriesser commented:
Let's postpone this PR until after #1517 is merged. Also will refresh the code with the latest work for multi-threaded debugging.
